go/worker/common/p2p/error/error.go

```diff
@@ -23,6 +57,40 @@ func IsPermanent(err error) bool {
 		// should actually be retried.
 		return false
 	}
-	_, ok := (err).(*backoff.PermanentError)
-	return ok
+	var e *backoff.PermanentError
+	return errors.As(err, &e)
+}
+
+// EnsurePermanent ensures an error will be correctly treated as (non-)permanent
+// by `cenkalti/backoff/v4`.
+//
+// Note: `IsPermanent` notion of a permanent error differs from the `cenkalti/backoff/v4`:
+// - it special cases `context.Canceled`
+// - it correctly handles wrapped permanent errors
+func EnsurePermanent(err error) error {
+	// TODO: file an issue upstream to make use of `errors.As`?
```

Comment:
Yeah that is a good idea, can you do it?

Reply:
yeah can do

Reply:
Thanks!

Reply:
done https://github.com/cenkalti/backoff/issues/107